docs/plugins.rst

```diff
@@ -1,260 +1,133 @@
+        def __init__(self, settings=Settings(), plugins=[ExamplePlugin(), WebHookExample(),
+                     MyPlugin()]
+        ):
 
+   If everything went as planned you can now start your bot, send the message "wake up" and expect to appropriate reply.
 
-    @respond_to('Give me (.*)')
-    def give_me(message, something):
-        message.reply('Here is %s' % something)
+Further configuration
+---------------------
 
+The below code snippets provide an insight into the functionality that can be added to the bot. For more in-depth examples,
+please refer to `./plugins/example.py` and `./plugins/webhook_example.py`.
 
-If you would like to have a command like 'stats' and 'stats start_date end_date', you can create reg ex like so:
+Implementing regular expression
+-------------------------------
 
 .. code-block:: python
 
-    from mmpy_bot.bot import respond_to
     import re
 
+    @listen_to('hi', re.IGNORECASE)
+    def hi(message):
+        message.reply('I can understand hi or HI!')
 
-    @respond_to('stat$', re.IGNORECASE)
-    @respond_to('stat (.*) (.*)', re.IGNORECASE)
-    def stats(message, start_date=None, end_date=None):
-        pass
-
-
-
-And add the plugins module to ``PLUGINS`` list of mmpy_bot settings, e.g. ``mmpy_bot_settings.py``:
-
-.. code-block:: python
-
-    PLUGINS = [
-        'mmpy_bot.plugins',
-        'devops.plugins',          # e.g. git submodule: domain:devops-plugins.git
-        'programmers.plugins',     # e.g. python package: package_name.plugins
-        'frontend.plugins',        # e.g. project tree: apps.bot.plugins
-    ]
-
-*For example you can separate git repositories with plugins on your team.*
-
-
-Attachment Support
-------------------
-
-.. code-block:: python
-
-    from mmpy_bot.bot import respond_to
-
-
-    @respond_to('webapi')
-    def webapi_reply(message):
-        attachments = [{
-            'fallback': 'Fallback text',
-            'author_name': 'Author',
-            'author_link': 'http://www.github.com',
-            'text': 'Some text here ...',
-            'color': '#59afe1'
-        }]
-        message.reply_webapi(
-            'Attachments example', attachments,
-            username='Mattermost-Bot',
-            icon_url='https://goo.gl/OF4DBq',
-        )
-        # Optional: Send message to specified channel
-        # message.send_webapi('', attachments, channel_id=message.channel)
-
-
-File Support
+    @listen_to('Give me (.*)')
+    async def give_me(self, message, something):
+        self.driver.reply_to(message, 'Here is %s' % something)
+
+Restrict messages to specific users
+----------------------------------
+
+    .. code-block:: python
+
+        @listen_to("^admin$", direct_only=True, allowed_users=["admin", "root"])
+        async def users_access(self, message: Message):
+            """Showcases a function with restricted access."""
+            self.driver.reply_to(message, "Access allowed!")
+
+Click support
```

Comment:
We should write a quick blurb here explaining the benefit of using Click?

Reply:
Agreed, but to be honest, I don't even know myself what it's for :smile:. @jneeven can you write a quick couple of lines about it?

Reply:
Click is amazing! I'll write something tomorrow :+1: